stdlib/atomic.mli

```diff
@@ -29,6 +29,10 @@ type !'a t
 (** Create an atomic reference. *)
 val make : 'a -> 'a t
 
+(** Create a cache line-aligned atomic reference. Use it over [make v] to
+    prevent false-sharing. *)
```

Comment:
Perhaps mention that memory requirements are increased?  Like 8x compared to a normal atomic reference.  If I'm making a list with an atomic reference at each cell, for instance, maybe I don't want the 8x size increase.